src/OrleansCodeGenerator/GrainReferenceGenerator.cs

```diff
@@ -191,7 +192,8 @@ private static MemberDeclarationSyntax[] GenerateInvokeMethods(Type grainType, A
                 var options = GetInvokeOptions(method);
 
                 // Construct the invocation call.
-                if (method.ReturnType == typeof(void))
+                var isOneWayTask = method.GetCustomAttribute<OneWayAttribute>() != null;
```

Comment:
Shouldn't we constrain it even further to only allow methods returning `Task` to be marked `[OneWay]`, and throw an error if a method returning `Task<T>` has the attribute? I mean to stress that no value is coming back.